java/util/src/main/java/com/google/protobuf/util/JsonFormat.java

```diff
@@ -151,9 +153,26 @@ public Printer includingDefaultValueFields() {
      * current {@link Printer}.
      */
     public Printer preservingProtoFieldNames() {
-      return new Printer(registry, includingDefaultValueFields, true);
+      return new Printer(registry, includingDefaultValueFields, true, omittingInsignificantWhitespace);
     }
-    
+
+    /**
+     * Create a new  {@link Printer}  that will omit all insignificant whitespace in the JSON output. This new Printer clones all other configurations from the current Printer.
+     * The new Printer clones all other configurations from the Insignificant whitespace is defined by the JSON spec as whitespace that appear between JSON structural elements:
```

Comment:
"The new Printer clones all other configurations from the" is duplicated from the above line.